runtime/io.c

```diff
@@ -77,28 +83,18 @@ static void channel_mutex_lock_default(struct channel *chan)
   caml_leave_blocking_section();
 }
 
-static void channel_mutex_unlock_default(struct channel *chan)
+CAMLexport void caml_channel_unlock(struct channel *chan)
 {
   caml_plat_unlock(&chan->mutex);
   last_channel_locked = NULL;
 }
 
-static void channel_mutex_unlock_exn_default(void)
+CAMLexport void caml_channel_cleanup_on_raise(void)
 {
   struct channel * chan = last_channel_locked;
-  if (chan != NULL && caml_channel_mutex_unlock != NULL)
-    caml_channel_mutex_unlock(chan);
+  caml_channel_unlock(chan);
```

Comment:
You need to test `chan != NULL` !!!

Reply:
Oops! I was in a hurry to remove the other NULL test and did not pay close attention. Fixed.